5493  - Mixpanel Growth Metrics

Miaplacidus opened the pull request:
This PR completes work for issue #5493, which asks for the following events to be logged to Mixpanel:

EVENTS FOR GROWTH TEAM TEAM PRIORITIZATION:
- [x]  Signed-in: total events, total unique users => # of users who visited commonwealth per quarter
- Create Community Page: frequency, weekly, monthly, quarterly
  - [x] Page Viewed: total number of attempts  
  - [x] Communities Created: successfully completed form
- [x] Joined Community: # of users becoming communities members, by community, over time

## Link to Issue
Closes: #5493 

## Description of Changes
- Log events around sign ins, the viewing and use of the create community page, and the number of users joining communities.

## Test Plan
- set the appropriate environment variables: NODE_ENV, MIXPANEL_TOKEN, and MIXPANEL_DEV_TOKEN
- on localhost, trigger the events described above and check that they are logged in Mixpanel (see the Events tab)

Miaplacidus commented:
@zakhap 
In regards to the final task of logging all of the join events, I've included every instance in which a user is added to a community. I did this so that, even when a user is attempting to join a community without having prior used its required wallet, their joining was still logged as a join event. My understanding is that logging this event also logs instances where the user does not necessarily explicitly ask to be added to a community, but is automatically added to a community. Let me know if this is acceptable. We could also log when users click on the 'Join Community' button for comparison.

jnaviask reviewed (changes requested):
- JOIN_COMMUNITY should be tracked at time of address verification rather than creation -- users can create arbitrary address but aren't considered "real" address until post-verification.
- Doesn't seem like magic logins are being counted as JOIN_COMMUNITY?

kurtisassad commented on packages/commonwealth/server/routes/linkExistingAddressToChain.ts:
```diff
@@ -164,13 +166,24 @@ const linkExistingAddressToChain = async (
+    const serverAnalyticsController = new ServerAnalyticsController();
+    serverAnalyticsController.track(
+      {
+        isCustomDomain: null,
```
Seems suspicious. Should be req.get('host') !== SERVER_URL I think.

Seems to be the same everywhere in the app. We should probably refactor them to be correct.

kurtisassad replied:
Or better yet, it could be added to body of the track method, then we could just remove this isCustomDomain from all the track calls options arguments.

Miaplacidus replied:
@kurtisassad 
Pardon me; I should have been more specific in my response. Your solution is very close, but needs a little bit of string wrangling. `req.get` returns just the domain, but (see `server/config.ts`) SERVER_URL is set to 'https`/`http` plus the domain.

Miaplacidus commented:
> * JOIN_COMMUNITY should be tracked at time of address verification rather than creation -- users can create arbitrary address but aren't considered "real" address until post-verification.
> * Doesn't seem like magic logins are being counted as JOIN_COMMUNITY?

@jnaviask 
* moved the join community logging after verification
* added logging to magic logins

Thanks for reviewing!
